app/forms/prior_authority/steps/additional_costs/detail_form.rb

```diff
@@ -0,0 +1,70 @@
+module PriorAuthority
+  module Steps
+    module AdditionalCosts
+      class DetailForm < ::Steps::BaseFormObject
+        include Rails.application.routes.url_helpers
+
+        PER_ITEM = 'per_item'.freeze
+        PER_HOUR = 'per_hour'.freeze
+
+        UNIT_TYPES = [PER_ITEM, PER_HOUR].freeze
+        attribute :id, :string
+        attribute :name, :string
+        attribute :description, :string
+        attribute :unit_type, :string
+        attribute :items, :integer
+        attribute :cost_per_item, :decimal, precision: 10, scale: 2
+        attribute :period, :time_period
+        attribute :cost_per_hour, :decimal, precision: 10, scale: 2
+
+        validates :name, presence: true
+        validates :description, presence: true
+        validates :unit_type, inclusion: { in: UNIT_TYPES, allow_nil: false }
+
+        validates :items, presence: true, numericality: { greater_than: 0 }, if: :per_item?
+        validates :cost_per_item, presence: true, numericality: { greater_than: 0 }, if: :per_item?
+
+        validates :period, presence: true, time_period: true, unless: :per_item?
+        validates :cost_per_hour, presence: true, numericality: { greater_than: 0 }, unless: :per_item?
+
+        def formatted_total_cost
+          NumberTo.pounds(per_item? ? item_cost : time_cost)
+        end
+
+        def method
```

Comment:
slight concern over the name `method` for a method. from a grepability point of view and name clash avoidance point of view perhaps `form_method`,`request_method` or `method_name`